calculator.js

```diff
@@ -1,28 +1,134 @@
-
 const exampleAdditionInput = {
   num1: 3,
   num2: 5,
   operation: 'add',
 }
 
+// const badInput = {
+//   num1: 'dog',
+//   num2: 5,
+//   operation: '*',
+// }
+
 const calculateUserInput = function (error, promptInput) {
-  console.log('This is the value of the promptInput variable that got passed in by prompt, after our prompt package collect user input', promptInput);
+  // console.log('This is the value of the promptInput variable that got passed in by prompt, after our prompt package collect user input', promptInput);
+
+  // Check for invalid data
+
+  if (isNaN(promptInput.num1)) {
+    console.log(`${promptInput.num1} is not a number.`)
+    return
+  }
+  
+  if (isNaN(promptInput.num2)) {
+    console.log(`${promptInput.num2} is not a number.`)
+    return
+  }
+
+  const ops = [ 'add', '+', 'subtract', '-', 'multiply', '*', 'divide', '/' ]
+
+  if (!(ops.includes(promptInput.operation))) {
+    console.log('unsupported operator')
+    return
+  }
+  
+  const num1 = parseInt(promptInput.num1)
+  const num2 = parseInt(promptInput.num2)
+  const op = promptInput.operation
+  
+  const calculate = function() {
+    switch (op) {
+      case 'add': 
+      case '+':
+        console.log(`${num1} + ${num2} = ${num1 + num2}`);
+        break;
+      case 'subtract': 
+      case '-':
+        console.log(`${num1} - ${num2} = ${num1 - num2}`)
+        break;
+      case 'multiply': 
+      case '*':
+        console.log(`${num1} * ${num2} = ${num1 * num2}`)
+        break;
+      case 'divide': 
+      case '/':
+        if (num2 === 0) {
+          console.log('You cannot divide by zero')
+          break;
+        }
+        console.log(`${num1} / ${num2} = ${num1 / num2}`)
+        break;
+    } 
```

Comment:
the `default` case acts like an `else` expression. If `op` doesn't hit any of the cases given, then it hits the `default` case like a catch-all.

```suggestion
    switch (op) {
      case 'add': 
      case '+':
        console.log(`${num1} + ${num2} = ${num1 + num2}`);
        break;
      case 'subtract': 
      case '-':
        console.log(`${num1} - ${num2} = ${num1 - num2}`)
        break;
      case 'multiply': 
      case '*':
        console.log(`${num1} * ${num2} = ${num1 * num2}`)
        break;
      case 'divide': 
      case '/':
        if (num2 === 0) {
          console.log('You cannot divide by zero')
          break;
        }
        console.log(`${num1} / ${num2} = ${num1 / num2}`)
        break;
      default:
        console.log('unsupported operator');
        break;
    } 
```